libs/vre/shared/app-string-literal/src/lib/multi-language-form.service.ts

```diff
@@ -1,43 +1,42 @@
 import { ChangeDetectorRef, Injectable } from '@angular/core';
-import { FormArray, FormBuilder, FormControl, FormGroup, ValidatorFn } from '@angular/forms';
+import { FormBuilder, ValidatorFn } from '@angular/forms';
 import { UserSelectors } from '@dasch-swiss/vre/shared/app-state';
 import { Store } from '@ngxs/store';
+import { DaschLanguage, isDaschLanguage } from './dash-language.type';
+import { MultiLanguageFormArray } from './multi-language-form-array.type';
 
 /** Component Provider used in combination with
  * MultiLanguageInputComponent and MultiLanguageTextareaComponent.
  * Those two components have the same business logic (here), but differ in their html.
  */
 @Injectable()
 export class MultiLanguageFormService {
-  readonly availableLanguages = ['de', 'fr', 'it', 'en', 'rm'];
+  readonly availableLanguages: DaschLanguage[] = ['de', 'fr', 'it', 'en', 'rm'];
   selectedLanguageIndex: number;
-  formGroup: FormGroup;
-  controlName: string;
+  formArray: MultiLanguageFormArray;
   validators: ValidatorFn[];
 
   inputValue: string | null = null;
 
-  get formArray() {
-    return this.formGroup.controls[this.controlName] as FormArray;
+  get selectedFormControl() {
+    const index = this.formArray
+      .getRawValue()
+      .findIndex(v => v.language === this.availableLanguages[this.selectedLanguageIndex]);
+    return this.formArray.controls[index];
   }
 
-  get selectedFormControl(): FormControl {
-    return (
-      this.formArray.controls.find(
-        control => control.value.language === this.availableLanguages[this.selectedLanguageIndex]
-      ) as FormControl
-    ) // TODO as FormControl = bad pattern, should be refactored, also form should be with Typed values
-      .get('value') as FormControl;
+  get selectedLanguageControl() {
+    return this.selectedFormControl.controls.value;
   }
 
   get invalidErrors() {
```

Comment:
check this getter usages. undefined is not expected in some places

Reply:
it is intended to be undefined, but the check is done before